bom/internal/pom.xml

```diff
@@ -0,0 +1,305 @@
+            </dependency>
+            <dependency>
+                <groupId>org.hibernate.search</groupId>
+                <artifactId>hibernate-search-integrationtest-mapper-orm-realbackend</artifactId>
+                <version>${project.version}</version>
+            </dependency>
+            <dependency>
+                <groupId>org.hibernate.search</groupId>
+                <artifactId>hibernate-search-integrationtest-mapper-pojo-standalone-realbackend</artifactId>
+                <version>${project.version}</version>
+            </dependency>
+            <dependency>
+                <groupId>org.hibernate.search</groupId>
+                <artifactId>hibernate-search-integrationtest-showcase-library</artifactId>
+                <version>${project.version}</version>
+            </dependency>
+            <dependency>
+                <groupId>org.hibernate.search</groupId>
+                <artifactId>hibernate-search-integrationtest-v5migrationhelper-engine</artifactId>
+                <version>${project.version}</version>
+            </dependency>
+            <dependency>
+                <groupId>org.hibernate.search</groupId>
+                <artifactId>hibernate-search-integrationtest-v5migrationhelper-orm</artifactId>
+                <version>${project.version}</version>
+            </dependency>
+            <dependency>
+                <groupId>org.hibernate.search</groupId>
+                <artifactId>hibernate-search-integrationtest-mapper-orm-coordination-outbox-polling</artifactId>
+                <version>${project.version}</version>
+            </dependency>
+            <dependency>
+                <groupId>org.hibernate.search</groupId>
+                <artifactId>hibernate-search-integrationtest-mapper-orm-batch-jsr352</artifactId>
+                <version>${project.version}</version>
+            </dependency>
+
+            <!-- Jackson: used by the Elasticsearch REST client and in tests (wiremock, ...) -->
+            <dependency>
+                <groupId>com.fasterxml.jackson.core</groupId>
+                <artifactId>jackson-core</artifactId>
+                <version>${version.com.fasterxml.jackson}</version>
+            </dependency>
+            <dependency>
+                <groupId>com.fasterxml.jackson.core</groupId>
+                <artifactId>jackson-annotations</artifactId>
+                <version>${version.com.fasterxml.jackson}</version>
+            </dependency>
+            <dependency>
+                <groupId>com.fasterxml.jackson.core</groupId>
+                <artifactId>jackson-databind</artifactId>
+                <version>${version.com.fasterxml.jackson.databind}</version>
+            </dependency>
+
+            <!-- Test -->
+            <dependency>
+                <groupId>junit</groupId>
+                <artifactId>junit</artifactId>
+                <version>${version.junit}</version>
+            </dependency>
```

Comment:
I think JUnit is used in the build of public modules as well... You probably need to include it in the `public-build` BOM?
Same with many dependencies below. In short anything that is used in unit (surefire) tests must be in `public-build`; anything that is used only in integration tests is safe to leave here.

This does sound complicated though... I wonder if it wouldn't be simpler to just have a public BOM + a build BOM, for both public and internal modules.
Or a public BOM + the rest of dependency management in one of the parent POMs (the root POM or the public parent POM).

Before you start frowning at me, I'll point out this is _not_ the same thing as your original proposal ;D

Reply:
😏 haha
yeah, that does make sense (about junit), on the other hand I'm surprised it didn't fail without it and all other dependencies... 
I've started with keeping dependencies in the root and adding the public bom. But if we want that public bom to have a parent of root or any other parent, it will pick up all of the remaining dependencies from root ...
If we don't add a parent, then we'd need to move the version properties to this bom (which I didn't want to do to keep things grouped in the root)...
then I added a build bom (it was called internal 😃) and used that to import in all 3 parents (public, internal, integrationtest) and it worked. But then I thought that we probably don't want to import an "internal bom", that most likely will not be published, into a public parent ... so that's what led me to create that weird bom with exclusions....

Nooow going back to it -- having that build/internal bom and use it for all parents would work, but then we'd just need to publish it too?

Reply:
I suspect we'll always need to publish whatever manages dependencies of our public artifacts, including test ones, otherwise the published POM for those artifacts won't be complete and this will lead to problems for consumers.

Unless we do like gradle and clean up the POMs before publishing them, but I don't know how to do that in Maven and don't want to make the setup too complicated...

Reply:
BTW we currently do publish all our dependency management, including for tests and stuff: it's in our root POM, which does get published.

So yeah, I'd say either:

* Put the "public" dependencies in the public BOM and the other ones in a "build" BOM.
* OR put the "public" dependencies in the public BOM and the other ones in the `parents/public` parent POM (and don't use that parent for the BOM but the root one instead, which will require *some* duplication I suppose)